crypto/s2n_hash.c

```diff
@@ -304,6 +308,7 @@ static int s2n_evp_hash_init(struct s2n_hash_state *state, s2n_hash_algorithm al
         GUARD_OSSL(EVP_DigestInit_ex(state->digest.high_level.evp.ctx, EVP_sha512(), NULL), S2N_ERR_HASH_INIT_FAILED);
         break;
     case S2N_HASH_MD5_SHA1:
+        notnull_check(state->digest.high_level.evp_md5_secondary.ctx);
```

Comment:
Isn't this the same check as in line 288 before the switch? Is there anyone it could be true there and false here?

Reply:
Good catch, I removed the extra check.